dotCMS/src/main/java/com/dotmarketing/startup/runonce/Task201014UpdateColumnsValuesInIdentifierTable.java

```diff
@@ -1,102 +1,166 @@
+public class Task201014UpdateColumnsValuesInIdentifierTable extends AbstractJDBCStartupTask {
 
     @Override
     public boolean forceRun() {
         return true;
     }
 
     @Override
-    public void executeUpgrade() throws DotDataException, DotRuntimeException {
+    public String getPostgresScript() {
 
-        final DotConnect dotConnect = new DotConnect()
-                .setSQL(DbConnectionFactory.isPostgres() ? UPDATE_CONTENTLETS_POSTGRESQL
-                        : UPDATE_CONTENTLETS);
-        dotConnect.loadResult();
+        final StringBuilder query = new StringBuilder();
 
-        // update owner, create_date and asset_subtype for folders
-        dotConnect.setSQL(DbConnectionFactory.isPostgres() ? UPDATE_FOLDER_POSTGRESQL
-                : UPDATE_FOLDER);
-        dotConnect.loadResult();
+        //update templates
+        query.append(getQueryToUpdateNonContentletsPostgres("template"));
 
-        // update owner, create_date and asset_subtype for containers
-        dotConnect.setSQL(DbConnectionFactory.isPostgres() ? UPDATE_CONTAINERS_POSTGRESQL
-                : UPDATE_CONTAINERS);
-        dotConnect.loadResult();
+        //update containers
+        query.append(getQueryToUpdateNonContentletsPostgres("dot_containers"));
 
-        // update owner, create_date and asset_subtype for templates
-        dotConnect.setSQL(DbConnectionFactory.isPostgres() ? UPDATE_TEMPLATES_POSTGRESQL
-                : UPDATE_TEMPLATES);
-        dotConnect.loadResult();
+        //update links
+        query.append(getQueryToUpdateNonContentletsPostgres("links"));
 
-        // update owner, create_date and asset_subtype for links
-        dotConnect
-                .setSQL(DbConnectionFactory.isPostgres() ? UPDATE_LINKS_POSTGRESQL : UPDATE_LINKS);
-        dotConnect.loadResult();
+        //update folders
+        query.append(getQueryToUpdateNonContentletsPostgres("folder"));
 
+        //update contentlets
+        query.append("update identifier set owner=mod_user, create_date=idate, asset_subtype=velocity_var_name from\n")
+                .append("(select distinct temp.identifier myID, temp.mod_user, inode.idate, struc.velocity_var_name  from contentlet temp, inode, structure struc,\n")
+                .append("(select identifier, MIN(idate) idate from contentlet, inode \n")
+                .append("where inode.inode=contentlet.inode group by identifier) custom_select  \n")
+                .append("where temp.identifier=custom_select.identifier\n")
+                .append("and inode.inode=temp.inode and inode.idate=custom_select.idate \n")
+                .append("and temp.structure_inode = struc.inode) my_query\n")
+                .append(" where  id=myID;\n");
+
+
+        return query.toString();
+    }
+
+    @Override
+    public String getMySQLScript() {
+        final StringBuilder query = new StringBuilder();
+
+        //update templates
+        query.append(getQueryToUpdateNonContentletsMySQL("template"));
```

Comment:
Codacy found an issue: [StringBuffer (or StringBuilder).append is called consecutively without reusing the target variable.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6393742)